src/main/java/net/sf/jabref/importer/CustomImporter.java

```diff
@@ -0,0 +1,137 @@
+/*
+ Copyright (C) 2005-2015 Andreas Rudert, Oscar Gustafsson extracted from CustomImportList
+
+ All programs in this directory and
+ subdirectories are published under the GNU General Public License as
+ described below.
+
+ This program is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2 of the License, or (at
+ your option) any later version.
+
+ This program is distributed in the hope that it will be useful, but
+ WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
+ USA
+
+ Further information about the GNU GPL is available at:
+ http://www.gnu.org/copyleft/gpl.ja.html
+  Copyright (C) 2005-2014 JabRef contributors.
+
+*/
+
+package net.sf.jabref.importer;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+
+import net.sf.jabref.importer.fileformat.ImportFormat;
+
+/**
+ * Object with data for a custom importer.
+ *
+ * <p>Is also responsible for instantiating the class loader.</p>
+ */
+public class CustomImporter implements Comparable<CustomImporter> {
+
+    private String name;
+    private String cliId;
+    private String className;
+    private String basePath;
```

Comment:
Because there are set-methods for them. (And Eclipse does it automatically if possible.)